packages/compiler-cli/src/ngtsc/typecheck/src/expression.ts

```diff
@@ -119,7 +121,8 @@ class AstTranslator implements AstVisitor {
     // interpolation's expressions. The chain is started using an actual string literal to ensure
     // the type is inferred as 'string'.
     return ast.expressions.reduce(
-        (lhs, ast) => ts.createBinary(lhs, ts.SyntaxKind.PlusToken, this.translate(ast)),
+        (lhs, ast) =>
+            ts.createBinary(lhs, ts.SyntaxKind.PlusToken, ts.createParen(this.translate(ast))),
```

Comment:
I suppose this is also done for diagnostics, so this could become

```suggestion
            ts.createBinary(lhs, ts.SyntaxKind.PlusToken, wrapForDiagnostics(this.translate(ast))),
```